LibCarla/source/carla/streaming/detail/tcp/ServerSession.cpp

```diff
@@ -102,7 +114,7 @@ namespace tcp {
       boost::asio::async_write(
           _socket,
           message->GetBufferSequence(),
-          boost::asio::bind_executor(_strand, handle_sent));
+          handle_sent);
```

Comment:
This is problematic, the `handle_sent` method cannot run concurrently as it is now, you would need to change it, call `Close()` instead of `CloseNow()` and make `_is_writing` atomic.